### DISCUSSION TITLE
Style.BasedOn supported?

### DISCUSSION THREAD

**Discussion by MihaMarkic** [AvaloniaUI/Avalonia#9339] [Q&A]
Hi guys,

I don't think the Style.BasedOn is currently supported. Is this something that's planned or how would one go with coding styles based on common parent? Probably just using Selector, I guess.

**Comment by timunie** [comment]
Hey @MihaMarkic,

in 11.0 we will have `ControlThemes`. I think they can be on other `ControlThemes`. Styles as they are implemented currently will still work. @maxkatz6 may correct me if I'm wrong.

**Reply by MihaMarkic** [reply]
Interesting. Do you have any docs/info about ControlThemes?

**Reply by timunie** [reply]
Just see latest source code for now. Examples in FluentTheme or SimpleTheme

**Reply by mainw** [reply]
Hello, will you add this mechanism?

**Reply by rabbitism** [reply]
> Hello, will you add this mechanism?

Already available.

**Comment by AnotherRedcoat** [comment]
While you wait for ControlThemes, note that the Classes property can take a space-separated list, so a base-style plus a custom-style can be applied.

```
        <Window.Styles>
                 <Style Selector="Button.baseButton">
			<Setter Property="BorderThickness" Value="5" />
			<Setter Property="Foreground" Value="White" />
			<Setter Property="FontWeight" Value="Bold" />
			<Setter Property="FontSize" Value="20" />
			<Setter Property="MinWidth" Value="100" />
			<Setter Property="Margin" Value="0 0 5 0" />
			<Setter Property="HorizontalContentAlignment" Value="Center"/>
		</Style>
		<Style Selector="Button.commandButton">
			<Setter Property="Background" >
				<Setter.Value>
					<LinearGradientBrush StartPoint="0%,0%" EndPoint="0%,100%">
						<GradientStop Color="#de7c26" Offset="0.00" />  
						<GradientStop Color="#de7c26" Offset="0.18" />  <!-- HSL 28, 74%, 51% -->
						<GradientStop Color="#917155" Offset="0.19" />  <!-- HSL 28, 26%, 45% -->						
						<GradientStop Color="#666666" Offset="0.70" />  <!-- HSL 0, 0%, 40% -->
					</LinearGradientBrush>
				</Setter.Value>
			</Setter>
		</Style>
	</Window.Styles>

        <Button Classes="baseButton commandButton" Click="OpenButtonClick">Open</Button>
```

**Reply by robloo** [reply]
@AnotherRedcoat Just a quick unrelated note: In 11.0 previews you can specify colors directly in HSL using the same format as HTML/CSS #7899

**Reply by AnotherRedcoat** [reply]
This is awesome, HSL is a more comfortable way to deal with colors. I tried it, got a lovely System.Xaml.XamlException then re-read what you said about the preview version. Looks like a lot of cool stuff coming.
